storage/maria/s3_func.c

```diff
@@ -1357,6 +1357,8 @@ static int s3_read_file_from_disk(const char *filename, uchar **to,
 
 err:
   error= my_errno;
+  if (!error)
```

Comment:
so what warning is this fixing?

Reply:
Coverity found a case where my_malloc could fail with no code in `my_error` (off the top of my head it was an OOM case), which leads to a `NULL` dereference elsewhere. I can dig out the Coverity ID for the hit if you need it as gets quite deep.

Reply:
Found it, ID 1520394, `s3_read_file_from_disk()` call in `aria_copy_to_s3()`, came up with a scenario where `alloc_block` contained `NULL` with no `my_error`. If you dig deep enough you can see that it was `my_malloc` doing this.

Although in retrospect this can only really happen if you tried to allocate SIZE_T_MAX. Which is only really possible if the S3 file is 4GB or more and the build is 32bit. An unlikely scenario. I'll remove this one.